ush/setup.py

```diff
@@ -1000,637 +981,195 @@ def get_location(xcs, fmt):
+                nhw=grid_params["NHW"],
+                run_task=False,
+                sfc_climo_fields=fixed_files["SFC_CLIMO_FIELDS"],
             )
-    else:
-        SFC_CLIMO_DIR = os.path.join(EXPTDIR, "sfc_climo")
-
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Set EXTRN_MDL_LBCS_OFFSET_HRS, which is the number of hours to shift
-    # the starting time of the external model that provides lateral boundary
-    # conditions.
-    #
-    # -----------------------------------------------------------------------
-    #
-    global EXTRN_MDL_LBCS_OFFSET_HRS
-    if EXTRN_MDL_NAME_LBCS == "RAP":
-        EXTRN_MDL_LBCS_OFFSET_HRS = EXTRN_MDL_LBCS_OFFSET_HRS or "3"
-    else:
-        EXTRN_MDL_LBCS_OFFSET_HRS = EXTRN_MDL_LBCS_OFFSET_HRS or "0"
-
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Set parameters according to the type of horizontal grid generation
-    # method specified.  First consider GFDL's global-parent-grid based
-    # method.
-    #
-    # -----------------------------------------------------------------------
-    #
-    global LON_CTR, LAT_CTR, NX, NY, NHW, STRETCH_FAC
-
-    if GRID_GEN_METHOD == "GFDLgrid":
-        grid_params = set_gridparams_GFDLgrid(
-            lon_of_t6_ctr=GFDLgrid_LON_T6_CTR,
-            lat_of_t6_ctr=GFDLgrid_LAT_T6_CTR,
-            res_of_t6g=GFDLgrid_NUM_CELLS,
-            stretch_factor=GFDLgrid_STRETCH_FAC,
-            refine_ratio_t6g_to_t7g=GFDLgrid_REFINE_RATIO,
-            istart_of_t7_on_t6g=GFDLgrid_ISTART_OF_RGNL_DOM_ON_T6G,
-            iend_of_t7_on_t6g=GFDLgrid_IEND_OF_RGNL_DOM_ON_T6G,
-            jstart_of_t7_on_t6g=GFDLgrid_JSTART_OF_RGNL_DOM_ON_T6G,
-            jend_of_t7_on_t6g=GFDLgrid_JEND_OF_RGNL_DOM_ON_T6G,
-            RUN_ENVIR=RUN_ENVIR,
-            VERBOSE=VERBOSE,
-        )
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Now consider Jim Purser's map projection/grid generation method.
-    #
-    # -----------------------------------------------------------------------
-    #
-    elif GRID_GEN_METHOD == "ESGgrid":
-        grid_params = set_gridparams_ESGgrid(
-            lon_ctr=ESGgrid_LON_CTR,
-            lat_ctr=ESGgrid_LAT_CTR,
-            nx=ESGgrid_NX,
-            ny=ESGgrid_NY,
-            pazi=ESGgrid_PAZI,
-            halo_width=ESGgrid_WIDE_HALO_WIDTH,
-            delx=ESGgrid_DELX,
-            dely=ESGgrid_DELY,
-        )
-    #
-    # -----------------------------------------------------------------------
-    #
-    # Otherwise
-    #
-    # -----------------------------------------------------------------------
-    #
-    else:
-        grid_params = {
-            "LON_CTR": LON_CTR,
-            "LAT_CTR": LAT_CTR,
-            "NX": NX,
-            "NY": NY,
-            "NHW": NHW,
-            "STRETCH_FAC": STRETCH_FAC,
-        }
```

Comment:
This use case should never be realized. If it is, there should be an error. That logic is now near line 602.